::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
: Copyright (c) 2017 Enea AB and others.
:
: All rights reserved. This program and the accompanying materials
: are made available under the terms of the Apache License, Version 2.0
: which accompanies this distribution, and is available at
: http://www.apache.org/licenses/LICENSE-2.0
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Mon, 9 May 2016 17:08:06 +0200
Subject: [PATCH] UX: Update bootstrap/target build time estimate.

While building for a different architecture (e.g. AArch64 on x86_64),
the bootstrap/target image build may take longer, due to latency
introduced by using qemu-user-static.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 .../fuel_bootstrap_cli/fuel_bootstrap/utils/bootstrap_image.py   | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/contrib/fuel_bootstrap/fuel_bootstrap_cli/fuel_bootstrap/utils/bootstrap_image.py b/contrib/fuel_bootstrap/fuel_bootstrap_cli/fuel_bootstrap/utils/bootstrap_image.py
index a535075..c7d67e9 100644
--- a/contrib/fuel_bootstrap/fuel_bootstrap_cli/fuel_bootstrap/utils/bootstrap_image.py
+++ b/contrib/fuel_bootstrap/fuel_bootstrap_cli/fuel_bootstrap/utils/bootstrap_image.py
@@ -169,9 +169,12 @@ def make_bootstrap(data):
     OSLO_CONF = cfg.CONF
     OSLO_CONF(opts, project='fuel-agent')
     mngr = manager.Manager(bootdata)
-    LOG.info("Build process is in progress. Usually it takes 15-20 minutes."
-             " It depends on your internet connection and hardware"
-             " performance.")
+    LOG.info("Build process is in progress. Usually it takes 15-20 minutes for"
+             " a native build (x86_64) and/or 30-45 minutes for each"
+             " cross-build (e.g. AArch64)."
+             " It depends on your internet connection, hardware performance"
+             " and selected bootstrap architecture(s)."
+             " This ISO supports AArch64 only.")
     mngr.do_mkbootstrap()

     return bootdata['bootstrap']['uuid'], bootdata['output']
